### PR TITLE
feat: Return summarized metrics to visualize a trial

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

As part of the larger development of the Metrics API, this adds an API endpoint where we only the data points needed to visually represent a logged metric.  Unlike previous endpoints which involved downsampling, it's based on an individual trial, you can request multiple metrics, and you can include training and validation metrics in a single API call.

Requests have the format:

```
/api/v1/trials/1200/summarize
?maxDatapoints=100
&endBatches=100000
&metricNames=loss,error
&metricType=METRIC_TYPE_UNSPECIFIED
```

## Test Plan

- Should have tests for different start / end batches
- Should have tests for a metric name appearing in training AND validation
- Should have tests for a metric name appearing in training XOR validation

## Commentary

Developing frontend / chart view

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/pages/TrialDetails/TrialChart.tsx [hunk 1]
```diff
@@ -29,36 +29,63 @@ const getChartMetricLabel = (metric: MetricName): string => {
...
+  //     const summ = await getTrialSummary({
+  //       endBatches: Math.ceil(max),
+  //       maxDatapoints: 30,
+  //       metricNames: metricNames,
+  //       startBatches: Math.floor(min),
+  //       trialId: trialId,
+  //     });
+  //     setTrialSummary(summ.metrics);
+  //   }
+  // };
```
Comment (hkang1) [inline webui/react/src/pages/TrialDetails/TrialChart.tsx]: **chore:** drop, if it remains a comment

File: webui/react/src/pages/TrialDetails/TrialChart.tsx [hunk 2]
```diff
@@ -81,7 +108,7 @@ const TrialChart: React.FC<Props> = ({
       key: trialId,
       legend: { show: false },
       plugins: [ tooltipsPlugin(), trackAxis() ],
-      scales: { x: { time: false }, y: { distr: scale === Scale.Log ? 3 : 1 } },
```
Comment (hkang1) [inline webui/react/src/pages/TrialDetails/TrialChart.tsx:84]: I see, so the numeric scale value was actually mapped to uPlot's scale?

File: webui/react/src/pages/TrialDetails/TrialChart.tsx (hunk 1 above)
Comment (mapmeld) [reply webui/react/src/pages/TrialDetails/TrialChart.tsx]: 👍  This resetZoom code is back

File: webui/react/src/pages/TrialDetails/TrialChart.tsx (hunk 2 above)
Comment (mapmeld) [reply webui/react/src/pages/TrialDetails/TrialChart.tsx]: originally I was borrowing that, in the end I kept this as-is and use MetricType on the API

**Comment by mapmeld** [comment]
Closing in favor of a new PR to support Single Trial chart + workloads

**Pull request closed without merging**